cmd/shell.go

```diff
@@ -276,6 +289,57 @@ func newStatsRegistry(addr string, logger blog.Logger) prometheus.Registerer {
 	return registry
 }
 
+// newOpenTelemetry sets up our OpenTelemetry tracing
+// It returns a graceful shutdown function to be deferred.
+func newOpenTelemetry(serviceName string, config OpenTelemetryConfig) func() {
+	r, err := resource.Merge(
+		resource.Default(),
+		resource.NewWithAttributes(
+			semconv.SchemaURL,
+			semconv.ServiceNameKey.String(serviceName),
+			semconv.ServiceVersionKey.String(core.GetBuildID()),
+		),
+	)
+	if err != nil {
+		FailOnError(err, "Could not create OpenTelemetry resource")
+	}
+
+	// Use a ParentBased sampler to respect the sample decisions on incoming
+	// traces, and TraceIDRatioBased to randomly sample new traces.
+	sampler := trace.TraceIDRatioBased(config.SampleRatio)
+	if !config.DisableParentSampler {
+		sampler = trace.ParentBased(sampler)
+	}
```

Comment:
Slightly awkward double-negative here; probably clearer to do
```suggestion
	sampler := trace.ParentBased(sampler)
	if config.DisableParentSampler {
		sampler = trace.TraceIDRatioBased(config.SampleRatio)
	}
```

Reply:
This suggestion doesn't quite work, because we want trace.ParentBased to wrap the TraceIDRatioBased one.  The suggestion overwrites it, instead of wrapping it.